src/nft/utils/urlParams.ts

```diff
@@ -0,0 +1,193 @@
+        trait_type & the other trait_value. This causes confusion since we don't know
+        whether user has selected two traits (cause we have two elements in our array)
+        or it's only one.
+
+        Using this block of code, we'll identify if that's the case.
+      */
+
+      if (clonedQuery['traits'].length === 2) {
+        const [trait_type, trait_value] = clonedQuery['traits'] as [string, string]
+        const fullTrait = `${trait_type}${trait_value}`
+        if (!fullTrait.includes(',')) {
+          if (
+            trait_type.startsWith('(') &&
+            !trait_type.endsWith(')') &&
+            trait_value.endsWith(')') &&
+            !trait_value.startsWith('(')
+          )
+            clonedQuery['traits'] = [`${trait_type},${trait_value}`]
+        }
+      }
+
+      clonedQuery['traits'] = clonedQuery['traits'].map((queryTrait: string) => {
+        const modifiedTrait = trimTraitStr(queryTrait.replace(/(")/g, ''))
+        const [trait_type, trait_value] = modifiedTrait.split(',')
+        const traitInStats = collectionStats.traits.find(
+          (item) => item.trait_type === trait_type && item.trait_value === trait_value
+        )
+
+        /*
+          For most cases, `traitInStats` is assigned. In case the trait
+          does not exist in our store, e.g "Number of traits", we have to
+          manually create an object for it.
+        */
+        const trait = traitInStats ?? { trait_type, trait_value, trait_count: 0 }
+
+        return trait as Trait
+      })
+    } catch (err) {
+      clonedQuery['traits'] = []
+    }
+
+    return clonedQuery
+  },
+}
+
+export const syncLocalFiltersWithURL = (state: CollectionFilters) => {
+  const urlFilterItems = [
+    'markets',
+    'maxPrice',
+    'maxRarity',
+    'minPrice',
+    'minRarity',
+    'traits',
+    'all',
+    'search',
+    'sort',
+  ] as const
+
+  const query: Record<string, any> = {}
+  urlFilterItems.forEach((key) => {
+    switch (key) {
+      case 'traits':
+        const traits = state.traits.map(({ trait_type, trait_value }) => `("${trait_type}","${trait_value}")`)
+
+        query['traits'] = traits
+        break
+
+      case 'all':
+        query['all'] = !state.buyNow
+        break
+
+      case 'sort':
+        query['sort'] = SortByPointers[state.sortBy]
+        break
+
+      default:
+        query[key] = state[key]
+        break
+    }
+  })
+
+  const modifiedQuery = urlParamsUtils.removeDefaults(query)
+
+  // Applying local state changes to URL
+  const url = window.location.href.split('?')[0]
+  const stringifiedQuery = qs.stringify(modifiedQuery, { arrayFormat: 'comma' })
+
+  // Using pushState on purpose here. router.push() will trigger re-renders & API calls.
+  window.history.pushState({}, ``, `${url}${stringifiedQuery && `?${stringifiedQuery}`}`)
+}
+
+export const applyFiltersFromURL = (location: Location, collectionStats: GenieCollection) => {
```

Comment:
on genie the url filters had a useeffect where they would be cleared on a change of pathname that we should add back, changing pages for me is maintaining the filters